scipy/stats/_distn_infrastructure.py

```diff
@@ -1254,12 +1254,12 @@ def entropy(self, *args, **kwds):
         place(output, cond0, self.vecentropy(*goodargs) + log(goodscale))
         return output
 
-    def moment(self, n, *args, **kwds):
+    def moment(self, order=None, *args, **kwds):
```

Comment:
Ultimately, we want the argument `n` to be renamed `order`. This requires a deprecation cycle during which:
1. we need to raise a deprecation warning when `n` is used as a keyword argument to refer to the first positional argument of `moment`. 
2. we allow the use of keyword argument `order` to be used instead of the first positional argument of `moment`.

In hindsight, rather than changing the name of the argument `n` to `order` and looking for `n` in `kwds`, it may have been easier to have left `n` alone and looked for `order` in `kwds`. The reason I didn't do this is that during the deprecation cycle we would still have the name collision, and the `check_dist_keyword_names` test in `refguide_check.py` would fail. However, I could have added some sort of exceptions to the rules there.

Reply:
Since we are doing this, I would also take the opportunity to also put these as keyword only. Or was this on purpose?

Reply:
This PR has been very, very careful to preserve backwards compatibility (otherwise this would have been much easier). So these are keyword or positional on purpose.

Reply:
What I suggest here is `def moment(self, *args, order=None, **kwds)`. It would preserve backward compatibility while forcing new code to use a positional only argument.

Reply:
Given the complexity of this PR already and the fact that few (if any?) `rv_generic` methods have keyword-only restriction, can we leave this alone? Perhaps if the suggestion had been made when this work started , it could have been accomodated. At this point, it is beyond the scope of what I set out to do.

Reply:
Sure ok 👌